src/lwc/geListView/geListView.js

```diff
@@ -374,64 +383,96 @@ export default class geListView extends LightningElement {
+            let referenceField = this.handleReferenceTypeFields(fieldDescribe);
+
+            if (isNotEmpty(referenceField)) {
+                columnEntry.fieldApiName = referenceField;
+            }
 
-            // Special case for relationship references e.g. 'CreatedBy.Name'
-            // so we can display the Name property of the reference in the table.
+            // // Special case for relationship references e.g. 'CreatedBy.Name'
+            // // so we can display the Name property of the reference in the table.
             if (columnEntry.fieldApiName.includes(`.${NAME}`)) {
-                fieldApiName = columnEntry.fieldApiName.split('.')[0];
+               fieldApiName = columnEntry.fieldApiName.split('.')[0];
             }
 
             // Need to convert types derived from schema to types useable by lightning-datable
             const types = {
                 'double': 'number',
                 'datetime': 'date',
-                'date': 'date-local'
+                'date': 'date-local',
+                'reference': URL,
+                'string': columnEntry.isNameField ? URL : 'string'
             };
-            const convertedType = types[fieldDescribe.dataType.toLowerCase()];
 
             columnEntry.fieldName = fieldApiName;
+
+            const convertedType = types[fieldDescribe.dataType.toLowerCase()];
+
             columnEntry.type = convertedType ? convertedType : fieldDescribe.dataType.toLowerCase();
 
-            this.handleTypesAttribute(columnEntry, fieldApiName);
+            columnEntry = this.handleTypesAttribute(columnEntry, fieldApiName);
 
-            this.columns = [...this.columns, columnEntry];
-            this.columnEntriesByName[columnEntry.fieldName] = columnEntry;
+            _columns.push(columnEntry)
         }
+
+        return _columns;
+    }
+
+
+    /*******************************************************************************
+    * @description Set json datable columns once all columns are built to improve rendering performance
+    *
+    * @param {array} columnEntries: List of json columns
+    */
+    setDatatableColumns(columnEntries) {
+        let _columnEntriesByName = {};
+
+        columnEntries.forEach(column => {
+            _columnEntriesByName[column.fieldName] = column;
+        });
+
+        this.columnEntriesByName = _columnEntriesByName;
+        this.columns = columnEntries;
     }
 
+
+
     /*******************************************************************************
     * @description Method checks to see if the provided field is a reference and
     * adjusts the relevant column entry properties as needed.
     *
     * @param {object} fieldDescribe: Field describe from the schema.
     * @param {object} columnEntry: A column header entry for lightning-datatable.
     */
-    handleReferenceTypeFields(fieldDescribe, columnEntry) {
+    handleReferenceTypeFields(fieldDescribe) {
```

Comment:
I would probably rename this at this point. We're shifting towards using the 'handleX' prefix for methods that receive some kind of an event. Maybe something like `getComputedReferenceFieldTypeApiName`.

Reply:
done!